### PR TITLE
Split guild to make easier to follow

### PR DESCRIPTION
**Pull request by jsturtevant**
Signed-off-by: James Sturtevant <jstur@microsoft.com>

**Reason for PR**:

The guide had multiple CNI providers in the one doc and was hard to follow. This splits it out so it is easier to follow.

**Issue Fixed**:

Issue #

**Requirements**

- [ ] Sqaush commits 
- [ ] Documentation
- [ ] Tests

**Notes**:

### CODE REVIEW

**Review by fabi200123 — APPROVED** [review]
This is much easier to follow and I believe it is better to have calico and flannel in 2 different files. Also, shall we remove the refference that this is work in progress and the reference to #239?
Also, I believe the PR name should be **"Split guide..."** not **"Split guild..."**

**Comment by jsturtevant** [comment]
> This is much easier to follow and I believe it is better to have calico and flannel in 2 different files. Also, shall we remove the refference that this is work in progress and the reference to #239?
> Also, I believe the PR name should be **"Split guide..."** not **"Split guild..."**

done

File: guides/guide-for-adding-windows-node.md [hunk 1]
```diff
@@ -191,37 +46,11 @@ Use the command that was given to you when you ran `kubeadm init` on a control p
...
-```
+### Networking Configuration
 
-If your new node is in the `NotReady` state it is likely because the calico image is still downloading. You can check the progress as before by checking on the calico pods in the `kube-system` namespace:
+Once you have a Linux-based Kubernetes control-plane node and a Windows node you are ready to choose a networking solution.
 
-```shell
-kubectl -n kube-system get pods -l app=calico
-```
+You have two choices:
```
Comment (marosset) [inline guides/guide-for-adding-windows-node.md:53]:
```suggestion
This guide offers two choices:
```

File: guides/calico.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Installing Calico
+
+This should be followed after adding your [Windows node to the cluster with kubeadm](guide-for-adding-windows-node.md#adding-windows-nodes).
+
+## Configuring Calico with Host Process containers
```
Comment (marosset) [inline guides/calico.md:5]:
```suggestion
## Configuring Calico with HostProcess containers
```

File: guides/calico.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+export CALICO_VERSION="v3.24.5"
+```
+
+Create the tigera-operator.yaml and the custom-resources.yaml for Calico: 
+```bash
+kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/$CALICO_VERSION/manifests/tigera-operator.yaml
+curl https://raw.githubusercontent.com/projectcalico/calico/$CALICO_VERSION/manifests/custom-resources.yaml -O
+# If IP CIDR differs from 192.168.0.0/16, make sure to modify it in custom-resources.yaml.
```
Comment (marosset) [inline guides/calico.md:31]: This is the cluster CIDR right?
We should specify here.

File: guides/calico.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+### Add Windows calico solution
+
+1. Add calico and kube-proxy daemonsets from sig-windows-tools/hostprocess/calico
+
+Now you can add Windows-compatible versions of calico and kube-proxy. In order to ensure that you get a compatible version of kube-proxy, you'll need to substitute the tag of the image. The following example shows usage for Kubernetes v1.25.3, but you should adjust the version for your own deployment.
+
+```bash
+curl -L https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/master/hostprocess/calico/kube-proxy/kube-proxy.yml | sed 's/KUBE_PROXY_VERSION/v1.25.3/g' | kubectl apply -f -
```
Comment (marosset) [inline guides/calico.md:56]: nit should we export `KUBE_PROXY_VERSION` and `CALICO_VERSION` env vars at the beginning so it is more obvious which values need to be set by the user?

File: guides/flannel.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
+# Installing Flannel
+
+This should be followed after adding your [Windows node to the cluster with kubeadm](guide-for-adding-windows-node.md#adding-windows-nodes).
+
+## Configuring Flannel with Host Process containers
```
Comment (marosset) [inline guides/flannel.md:5]:
```suggestion
## Configuring Flannel with HostProcess containers
```

File: guides/flannel.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+### Add Windows flannel solution
+
+1. Add Windows Flannel and kube-proxy DaemonSets
+
+Now you can add Windows-compatible versions of Flannel and kube-proxy. In order to ensure that you get a compatible version of kube-proxy, you'll need to substitute the tag of the image. The following example shows usage for Kubernetes v1.24.3, but you should adjust the version for your own deployment.
+
+```bash
+curl -L https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/master/hostprocess/flannel/kube-proxy/kube-proxy.yml | sed 's/KUBE_PROXY_VERSION/v1.25.3/g' | kubectl apply -f -
```
Comment (marosset) [inline guides/flannel.md:72]: Same comment as https://github.com/kubernetes-sigs/sig-windows-tools/pull/272/files#r1061714837

File: guides/calico.md (hunk 3 above)
Comment (jsturtevant) [reply guides/calico.md:31]: I think this is the pod CIDR.  I didn't modify this aspect of the guide. @fabi200123 do you know off hand?

File: guides/guide-for-adding-windows-node.md [hunk 7]
```diff
@@ -23,148 +18,11 @@ Your Kubernetes server must be at or later than version 1.22. To check the versi
...
-
 ### Joining a Windows worker node
 
 > **Note:** All code snippets in Windows sections are to be run in a PowerShell environment with elevated permissions (Administrator) on the Windows worker node.
 
-1. Install ContainerD, wins, kubelet, and kubeadm.
+1. Install ContainerD, kubelet, and kubeadm.
 
 ```PowerShell
 # Install ContainerD
```
[the remark this answers is not in the document]
Comment (profnandaa) [reply guides/guide-for-adding-windows-node.md:28]: **Note to self, I'll need to investigate this further**: within the VM, `curl.exe` kept on failing consistently (just drops connection). I had to switch to `Invoke-WebRequest -Uri $source -OutFile $destination` which worked okay. I'll need to investigate a little more on what could have been happening.

File: guides/guide-for-adding-windows-node.md [hunk 8]
```diff
@@ -23,148 +18,11 @@ Your Kubernetes server must be at or later than version 1.22. To check the versi
...
-
->  **Note** To find your version of kubernetes run the following command:
-> `kubeadm version`
-
 ### Joining a Windows worker node
 
 > **Note:** All code snippets in Windows sections are to be run in a PowerShell environment with elevated permissions (Administrator) on the Windows worker node.
 
-1. Install ContainerD, wins, kubelet, and kubeadm.
+1. Install ContainerD, kubelet, and kubeadm.
```
Comment (profnandaa) [inline guides/guide-for-adding-windows-node.md:25]: nit: add Git installation step before `.\PrepareNode.ps1` step since it's depended on it.

```
C:\Users\Administrator\k8s-setup\PrepareNode.ps1 : The term 'git' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.
At line:1 char:1
+ .\PrepareNode.ps1 -KubernetesVersion v1.26.0
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment (profnandaa) [reply guides/guide-for-adding-windows-node.md:25]: nit: make `.\PrepareNode.ps1` script idempotent; fails after running the second time when an initially created file is re-created:
```
----                 -------------         ------ ----
d-----         1/22/2023  10:28 PM                pki
New-Item : NewItemIOError
At C:\Users\Administrator\k8s-setup\PrepareNode.ps1:61 char:1
+ New-Item -path C:\var\lib\kubelet\etc\kubernetes\pki -type SymbolicLi ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceExists: (C:\var\lib\kubelet\etc\kubernetes\pki:String) [New-Item], IOException
    + FullyQualifiedErrorId : NewItemIOError,Microsoft.PowerShell.Commands.NewItemCommand
```

Actually it's only these 2 other places:
```
Registering kubelet service
Error creating service!
CreateService(): The specified service already exists.
```
----
```
Set parameter "DependOnService" for service "kubelet".
New-NetFirewallRule : Cannot create a file when that file already exists.
At C:\Users\Administrator\k8s-setup\PrepareNode.ps1:106 char:1
+ New-NetFirewallRule -Name kubelet -DisplayName 'kubelet' -Enabled Tru ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceExists: (MSFT_NetFirewallRule:root/standardcimv2/MSFT_NetFirewallRule) [New-NetF
   irewallRule], CimException
    + FullyQualifiedErrorId : Windows System Error 183,New-NetFirewallRule
```

**Comment by riverar** [comment]
Nudging this PR as it's getting stale and blocking important documentation.

File: guides/guide-for-adding-windows-node.md (hunk 8 above)
Comment (jsturtevant) [reply guides/guide-for-adding-windows-node.md:25]: > nit: add Git installation step before `.\PrepareNode.ps1` step since it's depended on it.

https://github.com/kubernetes-sigs/sig-windows-tools/pull/280 should resolve this

> nit: make .\PrepareNode.ps1 script idempotent; fails after running the second time when an initially created file is re-created:

let's do a follow up for this. This PR is just a copy/paste so it is easier to maintain

File: guides/guide-for-adding-windows-node.md [hunk 9]
```diff
@@ -3,14 +3,9 @@
...
-
-> The instructions and scripts in the directory DO NOT configure a CNI solution for Windows nodes running containerd.
-There is a work-in-progress PR to assist in this at https://github.com/kubernetes-sigs/sig-windows-tools/pull/239
-
 ## Before you begin
 
-Your Kubernetes server must be at or later than version 1.22. To check the version, enter `kubectl version`.
+Your Kubernetes cluster must be at or later than version 1.23. To check the version, enter `kubectl version`.
 
 - Obtain a [Windows Server 2019 license](https://www.microsoft.com/en-us/cloud-platform/windows-server-pricing) (or higher) in order to configure the Windows node that hosts Windows containers. If you are using VXLAN/Overlay networking you must have also have [KB4489899](https://support.microsoft.com/help/4489899) installed.
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply guides/guide-for-adding-windows-node.md:10]: you can use an evaluation copy for testing purposes

File: guides/guide-for-adding-windows-node.md [hunk 10]
```diff
@@ -3,14 +3,9 @@
...
 You can use Kubernetes to run a mixture of Linux and Windows nodes, so you can mix Pods that run on Linux on with Pods that run on Windows. This is a guide on how to register Windows nodes to your cluster.
 
-## Warning
-
-> The instructions and scripts in the directory DO NOT configure a CNI solution for Windows nodes running containerd.
-There is a work-in-progress PR to assist in this at https://github.com/kubernetes-sigs/sig-windows-tools/pull/239
-
 ## Before you begin
 
-Your Kubernetes server must be at or later than version 1.22. To check the version, enter `kubectl version`.
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply guides/guide-for-adding-windows-node.md:13]: > I just went through setting up the master node on a Hyper-V Ubuntu 22.04 VM and it was blood and tears;

Yes setting up a cluster from scratch isn't a trivial game.  There are tons of vendors and providers who attempt to solve this problem.

File: guides/calico.md (hunk 4 above)
Comment (jsturtevant) [reply guides/calico.md:56]: let's follow up with this, I this was a copy/paste of what is there currently to make it easier to follow/maintain

**Comment by jsturtevant** [comment]
/assign @marosset 
Since this is a straight copy paste, I would like to get it in so we can then make some of the valid suggestion made above

**Comment by Mik4sa** [comment]
What's the state here?

**Comment by jsturtevant** [comment]
will need to update since some changes went in.  If you want we can merge your other PR's and you can submit a PR like this and I will be able to merge it since it is a straight move to split the files out.

**Comment by Mik4sa** [comment]
@jsturtevant I just did so :).
